src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,41 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+
 import { Loader } from '../Loader';
+import { TodoModalProps } from './TodoModal.types';
+import { getUser } from '../../utils/getUser';
+
+export const TodoModal: React.FC<TodoModalProps> = ({
+  selectedTodo,
+  setSelectedTodo,
+}) => {
+  const [userName, setUserName] = useState('');
+  const [userIsLoading, setUserIsLoading] = useState(false);
+  const targetUserId = selectedTodo?.userId;
+
+  useEffect(() => {
+    if (targetUserId) {
+      setUserIsLoading(true);
+      setTimeout(() => {
```

Comment:
why do we need timeout here, looks redundant

Reply:
Didn't pass tests otherwise 
<img width="1129" alt="Screenshot 2024-07-03 at 17 37 39" src="https://github.com/mate-academy/react_dynamic-list-of-todos/assets/117541132/0c9be536-7b62-4f9a-8791-cd37cf44cd21">